Lib/importlib/_bootstrap_external.py

```diff
@@ -91,22 +92,35 @@ def _unpack_uint16(data):
     return int.from_bytes(data, 'little')
 
 
-def _path_join(*path_parts):
-    """Replacement for os.path.join()."""
-    return path_sep.join([part.rstrip(path_separators)
-                          for part in path_parts if part])
+if _MS_WINDOWS:
+    def _path_join(*path_parts):
+        """Replacement for os.path.join()."""
+        if not path_parts:
+            return ""
+        root = ""
+        path = []
+        for new_root, tail in map(_os._path_splitroot, path_parts):
+            if new_root.startswith(path_sep_tuple) or new_root.endswith(path_sep_tuple):
+                root = new_root.rstrip(path_separators) or root
+                path = [path_sep + tail]
+            else:
+                root = new_root or root
+                path.append(tail)
+        return root + path_sep.join(p.rstrip(path_separators) for p in path if p)
```

Comment:
If you keep the root slash in the first item of `path`, then it needs to be handled separately in case it's a single slash, i.e. `path = ['\\']`. For example:

```python
    path = [p.rstrip(path_separators) for p in path if p]
    if path == [''] and not root.startswith(path_sep_tuple):
        return root + path_sep
    return root + path_sep.join(path)
```

This allows resolving `_path_join(os.getcwd(), '/')` when the working directory is a drive path.